connectors/cli.py

```diff
@@ -87,6 +88,18 @@ def _parser():
     return parser
 
 
+async def _start_service(config, args, loop):
+    preflight = PreflightCheck(config)
+    for sig in (signal.SIGINT, signal.SIGTERM):
+        loop.add_signal_handler(sig, functools.partial(preflight.shutdown, sig))
+    if not await preflight.run():
```

Comment:
Done here:

https://github.com/elastic/connectors-python/blob/13f56dc139d2c3f1f42a97370e2cbe410c977c7b/connectors/cli.py#L95-L100

Instead of calling `shutdown`, I removed the signal handler, as `stop` will be done in the `finally` block of `PreflightCheck.run`:

https://github.com/elastic/connectors-python/blob/13f56dc139d2c3f1f42a97370e2cbe410c977c7b/connectors/preflight_check.py#L44-L47